### ISSUE TITLE
hon can't connect

### ISSUE THREAD

**Issue by tlf5000** [Andre0512/hon#51]
The hon login does not work.The application is being connected.Integration is not.

![2023-05-16_112337](https://github.com/Andre0512/hon/assets/62741564/4af968f9-9b8a-4e35-96de-fca01106987f)
![2023-05-16_112438](https://github.com/Andre0512/hon/assets/62741564/04adc501-7027-4370-bf69-05dc50908f52)

**Comment by Andre0512** [comment]
Hi, which type of appliances do you have?

**Comment by Andre0512** [comment]
Can you test the latest beta version `v0.8.0-beta.3`? I made the 'code' optional now and in the case that this isn't the only issue, then some device data should be logged now.

**Comment by tlf5000** [comment]
Great!Everything worked out, I got a device without objects.I would like to get his full support .What should I do?

**Comment by Pakstyle** [comment]
I installed it, ill test it in about 30~60min's, here is the log dump.
First thing i see is that the wheel is not working atm and the buttons underneath dont.
````Device Info:
data:
  additional_data:
  appliance:
    SK_Secondary: AC#selfClean56#2022-08-30
    applianceId: xx-xx-xx-xx-xx-xx#2022-07-01T17:29:15Z
    applianceModelId: 340
    applianceStatus: 1
    applianceTypeId: 11
    applianceTypeName: AC
    attributes:
      acuVersion: 06.26.00
      chipset: 1
      dictionaryId: 180
      eppParserSet: 1
      fwLabel: iotfw_epp
      lang: it-IT
      uiVersion: 06.26.00
    brand: haier
    connectivity: wifi|ble
    eepromId: 41
    eepromName: no_eeprom
    enrollmentDate: 2022-07-01T17:29:15.333Z
    firstEnrollment: False
    firstEnrollmentTBC: False
    fwVersion: 5.12.0
    id: 340
    lastUpdate: 2022-07-20T14:32:47Z
    macAddress: xx-xx-xx-xx-xx-xx
    modelName: AS35TADHRA-2
    nickName: Zolder
    purchaseDate: 2022-06-30T22:00:00.000Z
    sections:
      chatbot: True
      double_pairing_hidden: True
      epp_enabled: False
    series: tundra
    topics:
      publish:
      subscribe:
        - $aws/events/presence/disconnected/xx-xx-xx-xx-xx-xx
        - $aws/events/presence/connected/xx-xx-xx-xx-xx-xx
        - haier/things/xx-xx-xx-xx-xx-xx/event/appliancestatus/update
        - haier/things/xx-xx-xx-xx-xx-xx/event/discovery/update
    unitConfiguration: 1to1
  attributes:
    activity:
    commandHistory:
      command:
        ancillaryParameters:
          programFamily: [standard]
          remoteActionable: 1
          remoteVisible: 1
        applianceOptions:
        applianceType: AC
        attributes:
          channel: mobileApp
          energyLabel: 0
          origin: standardProgram
        commandName: startProgram
        device:
          appVersion: 1.53.7
          deviceModel: exynos9820
          mobileId: accef6d642a4b299
          mobileOs: android
          osVersion: 31
        macAddress: xx-xx-xx-xx-xx-xx
        parameters:
          10degreeHeatingStatus: 0
          ch2oCleaningStatus: 0
          cleaningTimeStatus: 0
          echoStatus: 0
          electricHeatingStatus: 0
          energySavePeriod: 15
          energySavingStatus: 0
          filterChangeStatusCloud: 0
          freshAirStatus: 0
          halfDegreeSettingStatus: 0
          healthMode: 0
          heatAccumulationStatus: 0
          humanSensingStatus: 0
          humidificationStatus: 0
          humiditySel: 30
          intelligenceStatus: 0
          lightStatus: 0
          lockStatus: 0
          machMode: 0
          muteStatus: 0
          onOffStatus: 1
          operationName: grSetDAC
          pm2p5CleaningStatus: 0
          pmvStatus: 0
          rapidMode: 0
          screenDisplayStatus: 1
          selfCleaning56Status: 0
          selfCleaningStatus: 0
          silentSleepStatus: 0
          specialMode: 0
          tempSel: 24
          tempUnit: 0
          voiceSignStatus: 0
          voiceStatus: 0
          windDirectionHorizontal: 0
          windDirectionVertical: 5
          windSensingStatus: 0
          windSpeed: 5
        timestamp: 2023-05-16T19:08:29.788Z
        transactionId: xx-xx-xx-xx-xx-xx_2023-05-16T19:08:29.788Z
    lastConnEvent:
      category: CONNECTED
      instantTime: 2023-05-16T07:09:32Z
      macAddress: xx-xx-xx-xx-xx-xx
      timestampEvent: 1684220972728
    parameters:
      10degreeHeatingStatus: 0
      acType: 0
      airQuality: 0
      ch2oCleaningStatus: 0
      ch2oValueIndoor: 0
      co2ValueIndoor: 0
      compressorCurrent: 51.10
      compressorFrequency: 0
      compressorStatus: 0
      debugEnabled: 0
      defrostStatus: 0
      echoStatus: 0
      electricHeatingStatus: 0
      electricHeatingStatusIndoor: 0
      energySavePeriod: 15
      energySavingStatus: 0
      errAckFlag: 0
      errors: 00
      expansionValveOpenDegree: 88
      fanStatusIndoor: 0
      fanStatusOutdoor: 2
      filterChangeStatusLocal: 0
      fourWayValveStatus: 2
      freshAirStatus: 0
      halfDegreeSettingStatus: 0
      healthMode: 0
      highTransRate: 1
      humanSensingStatus: 0
      humidificationStatus: 0
      humidityIndoor: 0
      humiditySel: 30
      intelligenceStatus: 0
      lightStatus: 0
      lockStatus: 0
      machMode: 1
      machineModeHK: 0
      muteStatus: 0
      onOffStatus: 0
      operationName: 
      operationSource: 1
      pm2p5CleaningStatus: 0
      pm2p5LevelIndoor: 0
      pm2p5ValueIndoor: 0
      pm2p5ValueOutdoor: 0
      pmvStatus: 0
      power: 0
      rapidMode: 0
      screenDisplayStatus: 1
      selfCleaningStatus: 0
      sensingResult: 0
      silentSleepStatus: 0
      specialMode: 0
      statusSyncRate: 1440
      stdTransRate: 3600
      tempAirOutdoor: 41
      tempCoilerIndoor: 21.00
      tempCoilerOutdoor: -64
      tempDefrostOutdoor: 13
      tempInAirOutdoor: -64
      tempIndoor: 20.50
      tempOutdoor: 12.00
      tempSel: 22.00
      tempUnit: 0
      totalElectricityUsed: 0
      totalWorkTime: 0
      transMode: 0
      vocValueIndoor: 0
      windDirectionHorizontal: 0
      windDirectionVertical: 0
      windSpeed: 5
    resultCode: 0
  settings:
    ancillaryParameters:
      programRules:
        selfCleaning56Status:
          $installationType:
            1to1:
              fixedValue: 0
              typology: fixed
            1to2:
              fixedValue: 0
              typology: fixed
            1toN:
              fixedValue: 0
              typology: fixed
        selfCleaningStatus:
          $installationType:
            1toN:
              fixedValue: 0
              typology: fixed
    custom:
      category: setParameters
    parameters:
      10degreeHeatingStatus: 0
      ch2oCleaningStatus: 0
      cleaningTimeStatus: 0
      echoStatus: 0
      electricHeatingStatus: 0
      energySavePeriod: 15
      energySavingStatus: 0
      filterChangeStatusCloud: 0
      freshAirStatus: 0
      halfDegreeSettingStatus: 0
      healthMode: 0
      heatAccumulationStatus: 0
      humanSensingStatus: 0
      humidificationStatus: 0
      humiditySel: 30
      intelligenceStatus: 0
      lightStatus: 0
      lockStatus: 0
      machMode: 0
      muteStatus: 0
      onOffStatus: 1
      operationName: grSetDAC
      pm2p5CleaningStatus: 0
      pmvStatus: 0
      rapidMode: 0
      screenDisplayStatus: 1
      selfCleaning56Status: 0
      selfCleaningStatus: 0
      silentSleepStatus: 0
      specialMode: 0
      tempSel: 22
      tempUnit: 0
      voiceSignStatus: 0
      voiceStatus: 0
      windDirectionHorizontal: 0
      windDirectionVertical: 5
      windSensingStatus: 0
      windSpeed: 5
  startProgram:
    ancillaryParameters:
      programRules:
        selfCleaning56Status:
          $installationType:
            1to1:
              fixedValue: 0
              typology: fixed
            1to2:
              fixedValue: 0
              typology: fixed
            1toN:
              fixedValue: 0
              typology: fixed
        selfCleaningStatus:
          $installationType:
            1toN:
              fixedValue: 0
              typology: fixed
      remoteActionable: 1
      remoteVisible: 0
    custom:
      program: iot_simple_start
    parameters:
      10degreeHeatingStatus: 0
      ch2oCleaningStatus: 0
      cleaningTimeStatus: 0
      echoStatus: 0
      electricHeatingStatus: 0
      energySavePeriod: 15
      energySavingStatus: 0
      filterChangeStatusCloud: 0
      freshAirStatus: 0
      halfDegreeSettingStatus: 0
      healthMode: 0
      heatAccumulationStatus: 0
      humanSensingStatus: 0
      humidificationStatus: 0
      humiditySel: 30
      intelligenceStatus: 0
      lightStatus: 0
      lockStatus: 0
      machMode: 0
      muteStatus: 0
      onOffStatus: 1
      operationName: grSetDAC
      pm2p5CleaningStatus: 0
      pmvStatus: 0
      rapidMode: 0
      screenDisplayStatus: 1
      selfCleaning56Status: 0
      selfCleaningStatus: 0
      silentSleepStatus: 0
      specialMode: 0
      tempSel: 24
      tempUnit: 0
      voiceSignStatus: 0
      voiceStatus: 0
      windDirectionHorizontal: 0
      windDirectionVertical: 5
      windSensingStatus: 0
      windSpeed: 5
  statistics:
    mostUsedPrograms:
      - count: 8
        prCode: None
        programName: PROGRAMS.AC.IOT_COOL
    mostUsedRecipes:
    programsCounter: 0
  stopProgram:
    parameters:
      10degreeHeatingStatus: 0
      ch2oCleaningStatus: 0
      cleaningTimeStatus: 0
      echoStatus: 0
      electricHeatingStatus: 0
      energySavePeriod: 15
      energySavingStatus: 0
      filterChangeStatusCloud: 0
      freshAirStatus: 0
      halfDegreeSettingStatus: 0
      healthMode: 0
      heatAccumulationStatus: 0
      humanSensingStatus: 0
      humidificationStatus: 0
      humiditySel: 30
      intelligenceStatus: 0
      lightStatus: 0
      lockStatus: 0
      machMode: 0
      muteStatus: 0
      onOffStatus: 0
      operationName: grSetDAC
      pm2p5CleaningStatus: 0
      pmvStatus: 0
      rapidMode: 0
      screenDisplayStatus: 1
      selfCleaning56Status: 0
      selfCleaningStatus: 0
      silentSleepStatus: 0
      specialMode: 0
      tempSel: 22
      tempUnit: 0
      voiceSignStatus: 0
      voiceStatus: 0
      windDirectionHorizontal: 0
      windDirectionVertical: 5
      windSensingStatus: 0
      windSpeed: 5
commands:
  settings:
    10degreeHeatingStatus:
      max: 1
      min: 0
      step: 1
    category:
      - setConfig
      - setParameters
    echoStatus:
      max: 1
      min: 0
      step: 1
    machMode:
      - 0
      - 1
      - 2
      - 4
      - 6
      - 0
    muteStatus:
      max: 1
      min: 0
      step: 1
    rapidMode:
      max: 1
      min: 0
      step: 1
    screenDisplayStatus:
      max: 1
      min: 0
      step: 1
    selfCleaningStatus:
      max: 1
      min: 0
      step: 1
    silentSleepStatus:
      max: 1
      min: 0
      step: 1
    tempSel:
      max: 30
      min: 16
      step: 1
    windDirectionVertical:
      - 2
      - 4
      - 5
      - 6
      - 7
      - 8
      - 5
    windSpeed:
      - 1
      - 2
      - 3
      - 5
      - 5
  startProgram:
    10degreeHeatingStatus:
      max: 1
      min: 0
      step: 1
    echoStatus:
      max: 1
      min: 0
      step: 1
    ecoMode:
      max: 1
      min: 0
      step: 1
    machMode:
      - 0
      - 1
      - 2
      - 4
      - 6
      - 0
    muteStatus:
      max: 1
      min: 0
      step: 1
    program:
      - iot_10_heating
      - iot_auto
      - iot_cool
      - iot_dry
      - iot_fan
      - iot_heat
      - iot_self_clean
      - iot_simple_start
    programFamily:
      - maintenance
      - health
      - [maintenance|health]
    rapidMode:
      max: 1
      min: 0
      step: 1
    remoteActionable:
      max: 1
      min: 0
      step: 1
    remoteVisible:
      max: 1
      min: 0
      step: 1
    screenDisplayStatus:
      max: 1
      min: 0
      step: 1
    selfCleaningStatus:
      max: 1
      min: 0
      step: 1
    silentSleepStatus:
      max: 1
      min: 0
      step: 1
    tempSel:
      max: 30
      min: 16
      step: 1
    windDirectionVertical:
      - 2
      - 4
      - 5
      - 6
      - 7
      - 8
      - 5
    windSpeed:
      - 1
      - 2
      - 3
      - 5
      - 5
  stopProgram:
    10degreeHeatingStatus:
      max: 1
      min: 0
      step: 1
    echoStatus:
      max: 1
      min: 0
      step: 1
    machMode:
      - 0
      - 1
      - 2
      - 4
      - 6
      - 0
    muteStatus:
      max: 1
      min: 0
      step: 1
    rapidMode:
      max: 1
      min: 0
      step: 1
    screenDisplayStatus:
      max: 1
      min: 0
      step: 1
    selfCleaningStatus:
      max: 1
      min: 0
      step: 1
    silentSleepStatus:
      max: 1
      min: 0
      step: 1
    tempSel:
      max: 30
      min: 16
      step: 1
    windDirectionVertical:
      - 2
      - 4
      - 5
      - 6
      - 7
      - 8
      - 5
    windSpeed:
      - 1
      - 2
      - 3
      - 5
      - 5
pyhOnVersion: 0.10.7````

**Comment by tlf5000** [comment]
7 objects were found.
![2023-05-17_110344](https://github.com/Andre0512/hon/assets/62741564/48923e45-5c94-4c39-b556-f78a9fd70a22)
The machine is turned off now.
When you click on "Show Device Info" this output:
```

```Device Info:
data:
  additional_data:
  appliance:
    applianceModelId: 216
    applianceStatus: 1
    applianceTypeName: WM
    attributes:
      acuVersion: 4.B.A
      chipset: ESP32D0WDQ5
      dictionaryId: 117
      lang: it-IT
      uiVersion: 6.A.0
    brand: candy
    eepromId: 95
    eepromName: 45320308
    enrollmentDate: 2020-10-03T07:17:55.434Z
    fwVersion: 5.12.0
    lastUpdate: 2022-03-02T18:31:31Z
    macAddress: xx-xx-xx-xx-xx-xx
    modelName: RO441286DWMC4-07
    nickName: Стиральная машина
    sections:
      ecoDelayStart: True
    topics:
      publish:
      subscribe:
        - $aws/events/presence/disconnected/xx-xx-xx-xx-xx-xx
        - $aws/events/presence/connected/xx-xx-xx-xx-xx-xx
        - haier/things/xx-xx-xx-xx-xx-xx/event/appliancestatus/update
        - haier/things/xx-xx-xx-xx-xx-xx/event/discovery/update
    waterHard: 2
  attributes:
    activity:
    commandHistory:
      command:
        ancillaryParameters:
          disTestOn: 0
          liquidDetergentDose: 2
          powderDetergentDose: 2
          programFamily: [dashboard]
          programType: W+D+S
          remoteActionable: 1
          remoteVisible: 1
          steamType: S
          suggestedLoadW: 3.5
          tempContribution: 1
        applianceOptions:
          opt1: prewash
          opt2: hygiene
          opt3: anticrease
          opt4: goodNight
          opt5: extraRinse1
          opt6: extraRinse2
          opt7: extraRinse3
          opt8: acquaplus
        applianceType: WM
        attributes:
          channel: mobileApp
          origin: standardProgram
        commandName: pauseProgram
        device:
          appVersion: 1.32.4
          deviceModel: exynos7870
          mobileId: f6bc9499a3f6a68b
          mobileOs: android
          osVersion: 29
        macAddress: xx-xx-xx-xx-xx-xx
        parameters:
          pause: 1
        timestamp: 2021-11-28T13:04:38.739Z
        transactionId: xx-xx-xx-xx-xx-xx_2021-11-28T13:04:38.739Z
      timestampAccepted: 2021-11-28T13:04:40.2Z
      timestampExecuted: 2021-11-28T13:04:40.6Z
    lastConnEvent:
      category: DISCONNECTED
      instantTime: 2023-05-16T04:53:26Z
      macAddress: xx-xx-xx-xx-xx-xx
      timestampEvent: 1684212806504
    parameters:
      acquaplus: 0
      anticrease: 0
      apsFreq: 62539
      apsOff: 0
      chartL: 0
      checkUpStatus: 0
      counterMV: 0
      delayTime: 0
      dirtyLevel: 0
      disTestOn: 0
      disTestRes: 0
      dryCotton120: 0
      dryCotton30: 0
      dryCotton60: 0
      dryCotton90: 0
      dryCottonExtra: 0
      dryCottonPA: 0
      dryCottonPS: 0
      dryLevel: 0
      drySynt120: 0
      drySynt30: 0
      drySynt60: 0
      drySynt90: 0
      drySyntExtra: 0
      drySyntPA: 0
      drySyntPS: 0
      dryTime: 0
      error: 00
      extraRinse1: 0
      extraRinse2: 0
      extraRinse3: 0
      fillR: 0
      goodNight: 0
      hygiene: 0
      lang: 16
      machMode: 0
      motS: 0
      nightWashStatus: 0
      ntcD: 7
      ntcW: 201
      numF: 0
      opt9: 0
      prCode: 37
      prCodeFav: 0
      prPhase: 0
      prPosition: 1
      prewash: 0
      program1: 96
      program10: 13
      program11: 173
      program12: 0
      program13: 11
      program14: 185
      program15: 0
      program16: 0
      program17: 0
      program18: 0
      program19: 5
      program2: 187
      program20: 3
      program21: 0
      program3: 8
      program4: 1
      program5: 43
      program6: 27
      program7: 104
      program8: 6
      program9: 21
      recipeId: 0
      remainingTimeMM: 3
      remoteCtrValid: 0
      spinSpeed: 1000
      steamLevel: 0
      t0R: 0
      t0W: 0
      tIW: 0
      temp: 30
      temp0to30: 646
      temp40: 182
      temp60to90: 55
      unbC: 0
      unbF: 0
    resultCode: 0
  statistics:
    drumCleaning:
      count: 883
      percentage: 883
      remaining: -783
      tot: 100
    filterCleaning:
      count: 883
      percentage: 883
      remaining: -783
      tot: 100
    lastCheckup: None
    loadingPercentage:
      - current: None
        date: 2023-05-16T04:07:11Z
        max: None
      - current: None
        date: 2023-05-16T04:07:11.0Z
        max: None
      - current: None
        date: 2023-05-15T16:30:16Z
        max: None
      - current: None
        date: 2023-05-15T16:30:16.5Z
        max: None
      - current: None
        date: 2023-05-14T11:14:43Z
        max: None
    mostUsedPrograms:
      - count: 78
        prCode: 4
        prPosition: 11
        programName: PROGRAMS.WD.SYNTHETIC_AND_COLOURED
        programType: W+D+S
      - count: 55
        prCode: 37
        prPosition: 1
        programName: PROGRAMS.WM_WD.SPECIAL_39
        programType: W+D
      - count: 37
        prCode: 12
        prPosition: 2
        programName: PROGRAMS.WD.MIX_AND_COLOUR_59
        programType: W+D+S
      - count: 33
        prCode: 37
        prPosition: 1
        programName: PROGRAMS.WD.SPECIAL_39
        programType: W+D
      - count: 28
        prCode: 10
        prPosition: 7
        programName: PROGRAMS.WM_WD.RAPID_30_MIN
        programType: W+D
    programsCounter: 455
    resultCode: 0
    temperatureUsage:
      temp0to30: 646
      temp40: 182
      temp60to90: 55
commands:
pyhOnVersion: 0.10.7```

```

**Comment by Andre0512** [comment]
Hey, I tried to fix it, please test `v0.8.0-beta.4`.

**Comment by tlf5000** [comment]
It works, but there are only 7 sensors, there is no control.Thank you very much!
